src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs

```diff
@@ -301,19 +301,14 @@ internal static TypedData ToRpcStringArray(this string[] arrString, bool include
             CollectionString collectionString = new CollectionString();
             foreach (string element in arrString)
             {
-                // Don't add null entries.("Add" method below will throw)
-                if (element is null)
+                // Empty string/null entries are okay to add based on includeEmptyEntries param value.
+                if (string.IsNullOrEmpty(element) && !includeEmptyEntries)
```

Comment:
This is a safe change because customers/worker needs to explicitly opt-in to this behavior by providing this capability (which we recently introduced to fix the issue @wesselkranenborg  [reported originally](https://github.com/Azure/azure-functions-dotnet-worker/issues/876)). Currently only dotnet worker is sending this capability (only when customer specify this behavior via worker options in [bootstrapping time](https://github.com/Azure/azure-functions-dotnet-worker/issues/876#issuecomment-1276646202))